File: python-procapi/Contracts/BackendMessage.py
from enum import Enum
from dataclasses import dataclass
from typing import Optional
from uuid import UUID


class BackendMessageType(Enum):
    Short = "short"
    Long = "long"

    def __str__(self):
        return self.value


@dataclass(frozen=True)
class BackendMessage:
    message_type: BackendMessageType
    data: str
    id: Optional[UUID]

File: rust_procapi/src/contracts/backend_message.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Serialize, Clone)]
pub(crate) enum BackendMessageType {
    Short,
    Long,
}

impl BackendMessageType {
    pub fn as_str(&self) -> &'static str {
        match *self {
            BackendMessageType::Short => "short",
            BackendMessageType::Long => "long"
        }
    }
}

#[derive(Serialize, Clone)]
pub(crate) struct BackendMessage {
    pub(crate) message_type: BackendMessageType,
    pub(crate) data: String,
    pub(crate) id: Option<Uuid>,
}